src/backend/api/objects.py

```diff
@@ -97,3 +143,464 @@ async def get_object_list(
         except s3.exceptions.ClientError:
             return None
     return res
+
+
+@router.head(
+    "/{bucket_name}/object",
+    responses=s3gw_client_responses(),
+)
+async def object_exists(
+    conn: S3GWClientDep, bucket_name: str, params: ObjectRequest
+) -> Response:
+    """
+    See
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/head_object.html
+    """
+    async with conn.conn() as s3:
+        await s3.head_object(
+            Bucket=bucket_name, Key=params.Key, VersionId=params.VersionId
+        )
+    return Response(content="", status_code=status.HTTP_200_OK)
+
+
+@router.get(
+    "/{bucket_name}/object",
+    response_model=Object,
+    responses=s3gw_client_responses(),
+)
+async def get_object(
+    conn: S3GWClientDep, bucket_name: str, params: ObjectRequest
+) -> Object:
+    """
+    See
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/head_object.html
+    """
+    async with conn.conn() as s3:
+        s3_res: HeadObjectOutputTypeDef = await s3.head_object(
+            Bucket=bucket_name, Key=params.Key, VersionId=params.VersionId
+        )
+        res = Object.parse_obj(
+            {
+                "Key": params.Key,
+                "Name": split_key(params.Key).pop(),
+                "Type": "OBJECT",
+                "Size": s3_res["ContentLength"],
+                **s3_res,
+            }
```

Comment:
This looks odd. Why aren't we using the model's constructor for this?

Reply:
I have to use this construct because of
```
  /home/vot/git/s3gw-ui/src/backend/api/objects.py:188:15 - error: Unable to match unpacked TypedDict argument to parameters
    No parameter named "DeleteMarker"
    No parameter named "AcceptRanges"
    No parameter named "Expiration"
    No parameter named "Restore"
    No parameter named "ArchiveStatus"
    No parameter named "ContentLength"
    No parameter named "ChecksumCRC32"
    No parameter named "ChecksumCRC32C"
    ... (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```